sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -224,7 +225,7 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
       nameInCache.length == name.length && nameInCache.zip(name).forall(conf.resolver.tupled)
     }
 
-    plan match {
+    EliminateSubqueryAliases(plan) match {
```

Comment:
I wonder why we keep `SubqueryAlias` when putting the logical plan into cache data.